### ISSUE TITLE
User keywords setup

### ISSUE THREAD

**Issue by manycoding** [robotframework/robotframework#2249]
I'm not sure why there's no [Setup] in user defined keywords.

Consider this:
I have a test suite which combines both data driven and regular test cases (just for the readability sake), and I use few templates for all test cases.
To keep it consistent I have to either add the keywords which I normally put in setup to template keywords and delete global Test Setup or split grouped test cases.

For example (note, this is just to show what I mean by combined test suite), here's what I have:
```
Suite Setup         Setup Suite
Suite Teardown      Teardown Suite
Test Setup          Open Tab
Test Teardown       Close Tab

*** Test Cases ***
Normal
    [Template]    Open File
    file.xml

Normal With Variable Input
    [Template]    Open File With Encoding
    file.xml    normal.txt
Normal With cp500 Variable Input
    [Template]    Open File With Encoding
    file.xml    normal.ebc    cp500
Normal With cp500 Variable Input
    [Template]    Open File With Encoding
    file.xml    normal.ebc_ibm    cp500

*** Keywords ***
Open File
    *keywords*

Open File With Encoding
    *keywords*
```
And here is what I want:
```
Suite Setup         Setup Suite
Suite Teardown      Teardown Suite

*** Test Cases ***
Normal
    [Template]    Open File
    file.xml

Normal With Variable Input
    [Template]    Open File With Encoding
    file.xml    normal.txt
    file.xml    normal.ebc    cp500
    file.xml    normal.ebc_ibm    cp500

*** Keywords ***
Open File
    [Setup]    Open Tab
    *keywords*
    [Teardown]    Close Tab

Open File With Encoding
    [Setup]    Open Tab
    *keywords*
    [Teardown]    Close Tab
 ```

**Comment by pekkaklarck** [comment]
Wouldn't your last example work as you want if you would just remove `[Setup]` from keywords?

**Comment by manycoding** [comment]
Ofcourse it will work and I have other workarounds. But setup keywords for me is just not really the part of tests.

I just want to know what justification is behind lack of [Setup] and, in the same time, presence of [Teardown] in keywords.

**Comment by pekkaklarck** [comment]
We added `[Teardown]` for keywords because there was a need to run a keyword at the end regardless of the keyword status. We decided not to add `[Setup]` because there is only a semantic difference between the first keyword and a setup.

On the test case level `[Setup]` makes sense because it allows overriding setup set with `Test Setup` setting. That setting is useful because it makes it possible to avoid using same first keyword with each test. There is nothing like that on the keyword level.

I still think adding keyword setup is not worth the effort. If you disagree, I would be happy to review pull requests.
